### DISCUSSION TITLE
Thoughts on using 3rd party API (for dislike counter and downloading a video)

### DISCUSSION THREAD

**Discussion by SuspiciousLookingOwl** [SuspiciousLookingOwl/youtubei#43] [Ideas]
As you might already know, YouTube with its infinite wisdom decided to remove dislike counter a while back. But we got https://returnyoutubedislike.com which does what it says, return YouTube dislike counter, it might not be accurate, but from my personal experience, it work quite well.

This got me thinking to integrate the [Return Youtube Dislike's API](https://returnyoutubedislikeapi.com/swagger/index.html) to get dislike counter of a video, for example:
```js
const video = await client.getVideo("videoId");
const dislikeCount = await video.getDislikeCount();
```
which will call `https://returnyoutubedislikeapi.com/votes?videoId=videoId` and return the dislike count.

But I'm not sure whether I should integrate any 3rd party API into this library, or just keep `Youtubei` as it is, which is to just interact with Youtube's innertube API.

We can also integrate with [ytdl-core](https://www.npmjs.com/package/ytdl-core) like this:
```js
const video = await client.getVideo("videoId");
const stream = video.download();
```
to make it easier to download or stream YouTube video.

---

@ConnorFTW @Hazmi35 @SushiBtw @mzrtamp I'm tagging you guys because seems like you are using `youtubei` actively on your project, maybe you guys can give some opinions on this.

**Comment by Hazmi35** [comment]
As long as it doesn't do that by default, I'm fine with it. 
So it wouldn't not make us create two request for two different API in the same time when we don't need the dislike count
